Made the following changes:

manikanta-osm opened the pull request:
1. Fixed warnings and notices
2. Change improper variable names
3. Placed hardcoded nonce lifetime variable in config file

Commit:
Made the following changes:
1. Fixed warnings and notices
2. Change improper variable names
3. Placed hardcoded nonce lifetime variable in config file

Commit:
Made the following changes:
1.Removed hook wp_enqueue_scripts
2.Formatted the code
3.Prefixed the global variables with UMM
4.Replaced multiple hyphens with -- for empty values

Commit:
Removed dependency of bootstrap and using jQuery for it's dependencies

Commit:
Made the following changes:
1. Removed externally loaded bootstrap files, glyphicons and jQuery
2. Fixed data table jerking issue and alignment issues
3. Fixed form ui issue in lower resolutions

AbijeetP commented on UserMetaManagement.php:
```diff
@@ -13,63 +13,55 @@
+	wp_enqueue_style('user-meta-style-css');
+	wp_enqueue_style (  'wp-jquery-ui-dialog');
+
+	wp_register_script("user-meta-datatable-js", UMM_PLUGIN_URL . '/js/datatables-min.js', array('jquery'));
```
why is this using double quotes?

AbijeetP commented on UserMetaManagement.php:
```diff
@@ -89,37 +81,37 @@ public function showMetaForm() {
+		}
+	    } else {
+		$user = null;
+	    } return ($this->parseTemplate(UMM_TEMPLATE . '/users-same-meta-information.php', $user));
```
why is this in the same line?

AbijeetP commented on UserMetaManagement.php:
```diff
@@ -178,21 +170,21 @@ public function getUserMetaDetails() {
      */
 
     public function updateUserMetaDetails() {
```
add documentation for the methods.

AbijeetP commented on UserMetaManagement.php:
```diff
@@ -205,42 +197,44 @@ public function updateUserMetaDetails() {
     // Fixing the nonce life time to 4 hours.
-     public function nonceLifeTime($time){ 
-        return 60*60*4;
+    public function nonceLifeTime($time) {
```
remove 4 hours from constant.

AbijeetP commented on UserMetaManagement.php:
```diff
@@ -13,63 +13,55 @@
+	    return;
+	}
+	
+	wp_register_style('user-meta-font-awesome-css', UMM_PLUGIN_URL . '/css/font-awesome.min.css');
```
please add `umm` infront of all registrations.

AbijeetP commented on UserMetaManagement.php:
```diff
@@ -13,63 +13,55 @@
+	add_action('admin_enqueue_scripts', array($this, 'enquerer'));
+	add_shortcode('show_meta_form', array($this, 'showMetaForm')); // Shortcode to view the form.
+	add_shortcode('specific_metakey', array($this, 'showAllUsersOfSpecificMetaKey')); // Shortcode to show the users of specific meta key combinations.
+	add_action('wp_ajax_meta_search', array($this, 'metaSearch')); // Action to search the meta key value combinations.
```
prefix!

AbijeetP commented on UserMetaManagement.php:
```diff
@@ -13,63 +13,55 @@
+	add_action('wp_ajax_meta_search', array($this, 'metaSearch')); // Action to search the meta key value combinations.
+	add_action('wp_ajax_get_user_meta_details', array($this, 'getUserMetaDetails'));  // Action to show  user meta information to the admin.
+	add_action('wp_ajax_update_user_meta_data', array($this, 'updateUserMetaDetails')); // Action to update the user meta information.
+	add_action('wp_ajax_delete_user_meta', array($this, 'deleteUserMetaDetails')); // Action to delete the user meta information.
```
comments on top of the statement.

AbijeetP commented on css/style.css:
```diff
@@ -161,18 +144,248 @@
+.table-striped > tbody > tr {
+    border-bottom: 1px #ddd;
+}
+thead {
```
add prefix for classes as well.

AbijeetP commented on css/style.css:
```diff
@@ -161,18 +144,248 @@
+.meta-form .form-inline .form-group {
+    display: inline-block;
+}
+.btn {
```
prefix for buttons also!

AbijeetP commented on js/script.js:
```diff
@@ -1,11 +1,56 @@
+	    designAlertDialogBoxes();
+	}
+    });   
+    $(".close-dialog").click(function (){
```
please add a new line before this.

AbijeetP commented on js/script.js:
```diff
@@ -1,11 +1,56 @@
         'scrollX': true,
-        "autoWidth": false
+        "autoWidth": false,
+	"columnDefs": [
```
code is not formatted.

AbijeetP commented on js/script.js:
```diff
@@ -42,10 +87,10 @@ var userMetaManager = (function () {
-            'security': myAjax.ajax_nonce,
-            'userId': userId,
-            'userMetaData': updatedMetaData
+            action: 'update_user_meta_data',
```
write some comments on top of the loops to explain what you are doing.

AbijeetP commented on js/script.js:
```diff
@@ -60,28 +105,45 @@ var userMetaManager = (function () {
+    /**
+     * Method to add styles to the alert message dialog boxes.
+     */
+    function designAlertDialogBoxes() {
```
please get the reference to the currently opened dialog and modify only that dialog.

AbijeetP commented on js/script.js:
```diff
@@ -123,7 +184,7 @@ var userMetaManager = (function () {
             "scrollY": "400px",
             "scrollCollapse": false,
-            sortable: false,
+             sortable: false,
```
please remove " from the code.

AbijeetP commented on template/users-same-meta-information.php:
```diff
@@ -5,11 +5,11 @@
-                <th>Last name</th>
-                <th>Email</th>
-                <th>Role</th>
+		<th>Email<i class="fa fa-fw fa-sort"></i></th>
```
format code.

manikanta-osm replied:
Okay Abijeet

AbijeetP reviewed (changes requested):
Please make the discussed changes.

Commit:
Worked on following suggestions:
1.Adding documentation for methods
2.Formatting the code.
3.Add plugin prefix constant for php and javascript variables
4.Add plugin prefix for css classes
5.Writing comments on top of the loops.
6.Get the reference to the currently opened dialog and modify only that dialog.
7.Don't use quotes for object properties.

AbijeetP commented on UserMetaManagement.php:
```diff
@@ -214,23 +216,23 @@ public function deleteUserMetaDetails() {
     // Function to create a user page in the user secion in the user seciton.
-    public function addUserPage() {
-	add_users_page('User Meta Management', 'User Meta Management', 'manage_options', 'user-meta-management', array($this, 'userMetaManagement'));
+    public function UMMAddUserPage() {
```
please dont prefix function names. They are inside a class ... so it is not an issue that they will ever clash.

Commit:
Reverted back the method names
Fixed dialog not opening issue after meta data is filtered.
